packages/access-contracts/src/contracts/DeploymentAccessManaged.sol

```diff
@@ -0,0 +1,167 @@
+    /*//////////////////////////////////////////////////////////////
+                            STATE VARIABLES
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Current controller address (starts as deployer, transitions to governance)
+    address public controller;
+
+    /*//////////////////////////////////////////////////////////////
+                                EVENTS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Emitted when controller is updated
+    event ControllerUpdated(
+        address indexed oldController,
+        address indexed newController
+    );
+
+    /// @notice Emitted when governance mode is permanently activated
+    event GovernanceModeActivated(address indexed governance);
+
+    /*//////////////////////////////////////////////////////////////
+                                ERRORS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Thrown when caller is not the deployment controller
+    error CallerIsNotDeploymentController(address caller);
+
+    /// @notice Thrown when invalid controller address is provided
+    error InvalidController(address controller);
+
+    /*//////////////////////////////////////////////////////////////
+                              CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Initializes the controller and access manager
+     * @param initialController Address of the initial controller (deployer)
+     * @param accessManager Address of the ProtocolAccessManager
+     */
+    constructor(
+        address initialController,
+        address accessManager
+    ) ProtocolAccessManaged(accessManager) {
+        if (initialController == address(0))
+            revert InvalidController(initialController);
+
+        controller = initialController;
+        emit ControllerUpdated(address(0), initialController);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                                MODIFIERS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Modifier for deployment-phase configuration (deployer only)
+     * @dev Only usable during deployment phase before governance transition
+     */
+    modifier onlyDeploymentController() {
+        if (!_isInDeploymentPhase()) {
+            revert CallerIsNotDeploymentController(msg.sender);
+        }
+        if (msg.sender != controller) {
+            revert CallerIsNotDeploymentController(msg.sender);
+        }
+        _;
+    }
+
+    /**
+     * @notice Modifier for configuration that can be done by either deployer or governance
+     * @dev During deployment: only controller can call
+     *      After governance transition: only governors can call
+     */
+    modifier onlyControllerOrGovernor() {
+        if (_isInDeploymentPhase()) {
```

Comment:
after some thinking, I'm not sure if that should be in the hands of the deployer. Similar to fleet and ark deployment - all of the deployed contracts have to go through the governance process - while in this case governance could not oppose.